exercises/simple-linked-list/example.scala

```diff
@@ -0,0 +1,94 @@
+import scala.reflect.ClassTag
+
+class Node[+T]
+case object Empty extends Node[Nothing]
+case class NonEmpty[T](value: T, var next: Node[T]) extends Node[T]
+
+trait SimpleLinkedList[T] {
```

Comment:
Maybe we should already provide this `trait` in a file `SimpleLinkedList.scala`?